ServerCore/FileManager.cs

```diff
@@ -136,4 +148,32 @@ private static CloudStorageAccount StorageAccount
             }
         }
     }
+
+    /// <summary>
+    /// Uploads files in the background and Saves them to the database when the upload is complete
+    /// </summary>
+    public class BackgroundFileUploader
+    {
+        PuzzleServerContext _context;
+
+        public BackgroundFileUploader(IServiceProvider serviceProvider)
+        {
+            IServiceScope newScope = serviceProvider.CreateScope();
+            _context = newScope.ServiceProvider.GetService<PuzzleServerContext>();            
+        }
+
+        public void CloneInBackground(ContentFile newFile, string fileName, int eventId, Uri sourceUri)
+        {
+            Task<Uri> t = FileManager.CloneBlobAsync(fileName, eventId, sourceUri);
+            t.ContinueWith((finishedTask) =>
+            {
+                newFile.Url = finishedTask.Result;
+                lock (_context)
+                {
+                    _context.ContentFiles.Add(newFile);
+                    _context.SaveChanges();
```

Comment:
Is it a problem that this makes hundreds of single property edits and commits them? Would it be better to batch up all ContentFile objects and make all the Add calls in one FinishBackgroundProcessing() call?

Reply:
Maybe add all clone tasks to a List<Task> and then gate the final step on some WhenAll() thing.

Reply:
Ordinarily, you'd be right that this is pretty inefficient. However, even with these fixes, the PD 2020 import still got killed before copying 100% of files, so gating saving all of the entries would have meant that no entries would have saved at all. This way, we save whatever progress we do make.